libraries/node-core-library/src/MapExtensions.ts

```diff
@@ -19,4 +19,21 @@ export class MapExtensions {
       targetMap.set(pair[0], pair[1]);
     }
   }
+
+  /**
+   * Converts a string-keyed map to an object.
+   * @remarks
+   * This function has the same effect as Object.fromEntries(map.entries())
+   * in supported versions of Node (\>= 12.0.0).
+   * @param map - The map that the object properties will be sourced from
+   */
+  public static toObject<TValue>(map: Map<string, TValue>): { [key: string]: TValue } {
+    return Array.from(map.entries()).reduce<{ [key: string]: TValue }>(
```

Comment:
💫 I feel like a simple `for` loop might be more readable and perhaps even faster than `reduce()`.

Reply:
You're probably not wrong :)